### PR TITLE
:tada: Source Amazon Ads add campaign negative keywords stream

### LINKED ISSUE #26337: Source Amazon Ads: new stream BudgetRules
### What area the feature impact?

Connectors

### Revelant Information

https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-rules/overview
in version 3, if a campaign's budget has been impacted by a budget rule, you will see an effectiveBudget field included in the campaign budget object returned by [POST /sp/campaigns/list](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#/Campaigns/ListSponsoredProductsCampaigns).

### LINKED ISSUE #26401: Source Amazon Ads: new stream Campaign Negative Keywords
### What area the feature impact?

Connectors

### Revelant Information

https://advertising.amazon.com/API/docs/en-us/sponsored-products/negative-targeting/keywords

### PR DESCRIPTION
**Pull request by midavadim**
## What
 Added Ads Campaign Negative Keywords stream

Issue: https://github.com/airbytehq/airbyte/issues/26401
Doc: https://advertising.amazon.com/API/docs/en-us/guides/sponsored-products/negative-targeting/keywords

Add sponsored_display_budget_rules stream

Issue: https://github.com/airbytehq/airbyte/issues/26337
Doc: https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi/prod#/BudgetRules/GetSDBudgetRulesForAdvertiser

## Recommended reading order
1. `y.python`

## 🚨 User Impact 🚨
No

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`7f21d8e2f2`](https://github.com/airbytehq/airbyte/pull/27482/commits/7f21d8e2f2b0c858ac66c82c51cf24d5725d99a1)) - ✅

⏲️  Total pipeline duration: 203 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5315371236)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Review by lazebnyi — CHANGES_REQUESTED** [review]
Please update PR description and fix conflicts

**Comment by midavadim** [comment]
> Please update PR description and fix conflicts

Let's process with review approval, there are a lot of continuous changes to Amazon Ads connector.

**Review by lazebnyi — CHANGES_REQUESTED** [review]
Please add link to issue and to provider doc with endpoint description and still need fix conflicts.

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`2bfb50e924`](https://github.com/airbytehq/airbyte/pull/27482/commits/2bfb50e92424cb0a4d2625d0491c0c7f8a0f5da4)) - ❌

⏲️  Total pipeline duration: 197 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-aamazon-ads-campaign-negative-keywords/1687522233/2bfb50e92424cb0a4d2625d0491c0c7f8a0f5da4/source-amazon-ads/2.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`642436999d`](https://github.com/airbytehq/airbyte/pull/27482/commits/642436999d3f767d3c01fb7d4f1854949f52e5b7)) - ✅

⏲️  Total pipeline duration: 179 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-aamazon-ads-campaign-negative-keywords/1687525463/642436999d3f767d3c01fb7d4f1854949f52e5b7/source-amazon-ads/2.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

File: airbyte-integrations/connectors/source-amazon-ads/Dockerfile [hunk 1]
```diff
@@ -14,5 +14,5 @@ ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
 
 
-LABEL io.airbyte.version=2.0.0
+LABEL io.airbyte.version=2.0.1
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-amazon-ads/Dockerfile:17]:
```suggestion
LABEL io.airbyte.version=2.1.0
```

File: docs/integrations/sources/amazon-ads.md [hunk 2]
```diff
@@ -94,6 +95,7 @@ Information about expected report generation waiting time you may find [here](ht
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                         |
 |:--------|:-----------|:---------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------|
+| 2.0.1   | 2023-06-19 | [25412](https://github.com/airbytehq/airbyte/pull/25412) | Add sponsored_product_campaign_negative_keywords stream                                                         |
```
Comment (lazebnyi) [inline docs/integrations/sources/amazon-ads.md:98]:
```suggestion
| 2.1.0   | 2023-06-19 | [25412](https://github.com/airbytehq/airbyte/pull/25412) | Add sponsored_product_campaign_negative_keywords stream                                                         |
```

**Review by lazebnyi — APPROVED** [review]
Approved, but need version change.

File: docs/integrations/sources/amazon-ads.md (hunk 2 above)
Comment (lazebnyi) [reply docs/integrations/sources/amazon-ads.md:98]: Need add stream to list in the doc.

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`393e626ae2`](https://github.com/airbytehq/airbyte/pull/27482/commits/393e626ae24592ed00762e0733c8899fed6e844f)) - ❌

⏲️  Total pipeline duration: 49 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-aamazon-ads-campaign-negative-keywords/1687529630/393e626ae24592ed00762e0733c8899fed6e844f/source-amazon-ads/2.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`bfd71d899e`](https://github.com/airbytehq/airbyte/pull/27482/commits/bfd71d899ecad4034f93206fc21a37449e3e75a3)) - ✅

⏲️  Total pipeline duration: 195 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-aamazon-ads-campaign-negative-keywords/1687529937/bfd71d899ecad4034f93206fc21a37449e3e75a3/source-amazon-ads/2.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Pull request merged**